dynet/nodes.cc

```diff
@@ -1275,6 +1275,24 @@ void NoBackprop::backward_dev_impl(const MyDevice & dev,
 DYNET_NODE_INST_DEV_IMPL(NoBackprop)
 
 template<class MyDevice>
+void FlipGradient::forward_dev_impl(const MyDevice & dev, const vector<const Tensor*>& xs, Tensor& fx) const {
+  fx.d = xs[0]->d;
+  fx.v = xs[0]->v;
+}
+
+template<class MyDevice>
+void FlipGradient::backward_dev_impl(const MyDevice & dev,
+                                   const vector<const Tensor*>& xs,
+                                   const Tensor& fx,
+                                   const Tensor& dEdf,
+                                   unsigned i,
+                                   Tensor& dEdxi) const {
+  // takes negative on backprop
+  dEdxi.tvec().device(*dev.edevice) = dEdf.tvec().unaryExpr(scalar_negative_op<float>());
```

Comment:
There's one major issue here, and one way to potentially make things easier. The major issue is that `backward_dev_impl` must accumulate gradients, so instead of `=` we would need `+=`. Separately from this, we might be able to just use `-=` instead of using `unaryExpr`. In summary, how about changing this line to:

    dEdxi.tvec().device(*dev.edevice) -= dEdf.tvec()